CMakeLists.txt

```
@@ -173,13 +173,17 @@ else()
 	set(INNOEXTRACT_HAVE_LZMA 0)
 endif()
 
-find_package(Boost REQUIRED COMPONENTS
+set(BOOST_REQUIRED_COMPONENTS
 	iostreams
 	filesystem
 	date_time
-	system
 	program_options
 )
+find_package(Boost REQUIRED COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
+if(Boost_MAJOR_VERSION EQUAL 1 AND Boost_MINOR_VERSION LESS 69)
+	list(APPEND BOOST_REQUIRED_COMPONENTS system)
+	find_package(Boost REQUIRED COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
+endif()
```

```suggestion
if(Boost_MAJOR_VERSION EQUAL 1 AND Boost_MINOR_VERSION LESS 69)
	list(APPEND BOOST_REQUIRED_COMPONENTS system)
endif()
find_package(Boost REQUIRED COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
```

Remove unnecessary line by reordering.

---

`Boost_MAJOR_VERSION`/`Boost_MINOR_VERSION` is set by the first `find_package` so change won't work.

---

Ah, ok. Doesn't knew that. :)